coderd/database/dbauthz/dbauthz.go

```diff
@@ -359,6 +376,12 @@ func AsSystemRestricted(ctx context.Context) context.Context {
 	return context.WithValue(ctx, authContextKey{}, subjectSystemRestricted)
 }
 
+// AsSystemReadProvisionerDaemons returns a context with an actor that has permissions
+// to read provisioner daemons.
+func AsSystemReadProvisionerDaemons(ctx context.Context) context.Context {
+	return context.WithValue(ctx, authContextKey{}, subjectSystemReadProvisionerDaemons)
+}
+
```

Comment:
self-review: I wanted to avoid sprinkling `dbauthz.AsSystemRestricted` everywhere, so I made a separate RBAC role for when we just wish to read provisioner daemons. I can remove this and switch back to `SystemRestricted` if folks prefer.